### DISCUSSION TITLE
Float pane / popup support

### DISCUSSION THREAD

**Discussion by milch** [ghostty-org/ghostty#3197] [Feature Requests, Ideas]
[Tmux](https://arc.net/l/quote/ffluycjy), [zellij](https://zellij.dev/news/floating-panes-tmux-mode/) (and soon [WezTerm](https://github.com/wez/wezterm/pull/5576)) have support to open a transient window within the current window. These can either act as a normal terminal or spawn a different command on launch. This has a few interesting use-cases:

- Quick scratch terminal to run a handful of commands while using your editor
- Use fzf to filter some list, then run a command with the result - e.g. picking a tmux session, quickly opening a specific file from our workspace folder, ...
- Open a TUI program like `lazygit`
- Run some frequently-run CLI command that is "system wide", e.g. `home-manager`, `darwin-rebuild`, `stow`, `chezmoi`, ...

Really there are a lot of use-cases this enables. The existing quick-term with the center option fulfills some of the same use-cases, but as far as I can see there is no way to spawn a command into the quick terminal or to exit it automatically when a command finishes (or create multiple, independent quick terminals). Some people also use e.g functionality built-in to editors like neovim for similar things, but that limits usage to happen from within the editor

---

Edit: If you are interested in this feature please use the upvote rather than posting a comment, unless you have specific notes on the implementation. This avoids sending a notification to anyone who is subscribed, and helps keep the notifications clean for the busy maintainers. Thanks!

**Comment by pandalec** [comment]
This would be nice. E.g. for an helix editor user like me, I built around helix a dozen snippets like you mentioned. They can be basically placed into two groups 'open X in tmux-popup' or 'open Y in tmux-window'. Within tmux I can search and run commands inside a specific window and create the window if it doesn't exist. Same vice versa, using yazi as a file manager which opens a file inside a running helix by sending `:o <file path>` within the current tmux session to the window with the name 'hx'.

**Comment by aleksandersumowski** [comment]
This could also be used to implement fuzzy search in the spirit of https://github.com/humeniuc/tmux-extrakto This is one functionality that I keep missing in other terminals/multiplexers since switching away from tmux.


That would be also one of the solutions to https://github.com/ghostty-org/ghostty/issues/189

**Comment by 0x3at** [comment]
This would be a great feature, definitely super high on my wish list.

**Reply by 00-kat** [reply]
Tried this cool little +1 button yet?
![A screenshot of the upvote button on the main post.](https://github.com/user-attachments/assets/43b21c9f-a324-40ce-9d35-972acba5302c)
[Octocat] came to be in my dreams and told me that if one thinks something is a great feature and super high on one's wish list for a project, pressing that button is better than adding a useless comment.  Apparently people subscribed wouldn't get an email too, though of course that *definitely* doesn't matter as much as pleasing Octocat.

I've unfortunately not had any opportunities to try it out, as I don't normally add useless comments such as your one; would you mind trying it out instead please?

Of course, had you brought something more useful to the conversation, your comment would have definitely been justified.

[Octocat]: https://github.com/octocat

**Comment by charlesvallieres** [comment]
Built this with Claude Code : https://github.com/ghostty-org/ghostty/compare/main...charlesvallieres:ghostty:popup

~/.config/ghostty/config
```
keybind = cmd+g=popup:lazygit
command-palette-entry = title:Lazygit,action:popup:lazygit

keybind = cmd+o=popup:tv dirs,60,80
command-palette-entry = title:TV,action:"popup:tv dirs,60,80"
```

Disclaimer : 
- Built with Claude Code Opus 4.6 1m
- I don't know Zig and/or macOS apps
- Not implemented with GTK
- The `command_no_wait` thing is probably weird and would require project knowledge
- ~~There are some quirks to fix like window resizing hiding the popup behind the main window~~
- There are probably some other bugs

Lacking the appropriate knowledge of the project and language I will not open PR. If anyone is willing to do the appropriate work, be my guest.

<img width="1796" height="1151" alt="Screenshot 2026-03-08 at 19 51 00" src="https://github.com/user-attachments/assets/2a499909-f6ea-4c11-887e-7865d7b663c9" />
